src/allowed_args.cpp

```diff
@@ -311,10 +312,19 @@ static void addGeneralOptions(AllowedArgs &allowedArgs, HelpMessageMode mode)
                     DEFAULT_PERSIST_MEMPOOL))
         .addArg("prune=<n>", requiredInt,
             strprintf(_("Reduce storage requirements by pruning (deleting) old blocks. This mode is incompatible with "
-                        "-txindex and -rescan. "
+                        "-txindex, -rescan, and -prunewithmask"
                         "Warning: Reverting this setting requires re-downloading the entire blockchain. "
                         "(default: 0 = disable pruning blocks, >%u = target size in MiB to use for block files)"),
                     MIN_DISK_SPACE_FOR_BLOCK_FILES / 1024 / 1024))
+        .addArg("prunewithmask={true,false,0,1}", optionalBool,
+            strprintf(_("Reduce storage requirements by pruning (deleting) old blocks. This mode is incompatible with "
+                        "-txindex, -rescan, and -prune "
```

Comment:
What happens if these flags are both used?  We should make sure no crash.

Reply:
i have reworked SetupPruning to address this concern. All pruning mode checks are now done in this function instead of during various different parts of init